src/app/manager-dashboard/reports/reports-detail.component.ts

```diff
@@ -420,4 +420,22 @@ export class ReportsDetailComponent implements OnInit, OnDestroy {
     this.dateFilterForm.controls.startDate.setValue(startDate || this.minDate);
     this.dateFilterForm.controls.endDate.setValue(endDate || this.today);
   }
+
+  filterByMember(filterValue, filterData?: []) {
+    let members = [];
+    const isSelectedMember = item => members.length === 0 ||
+      members.some(member => (member.userId === item.userId || member.userId.split(':')[1] === item.user));
+    if (filterValue !== 'All') {
+      this.couchService.findAll('teams', findDocuments({ teamId: filterValue._id, docType: 'membership' }))
+        .subscribe((teamMembers: any) => {
+          members = teamMembers;

```

Comment:
@singharpita This is a request to the database that will happen asynchronously to the rest of the function.  The function will return a value long before `members` is reassigned with the `teamMembers` from this request.